jupyterhub/orm.py

```diff
@@ -813,7 +814,18 @@ def new(
         else:
             assert service.id is not None
             orm_token.service = service
-        if expires_in is not None:
+        if expires_in:
+            if not isinstance(expires_in, numbers.Real):
+                raise TypeError(
+                    f"expires_in must be a positive integer or null, not {expires_in!r}"
+                )
+            expires_in = int(expires_in)
+            # tokens must always expire in the future
+            if expires_in < 1:
```

Comment:
```suggestion
            if expires_in < 0:
```

Reply:
This is nonzero on purpose - to prevent e.g. `0.5` from rounding down to 0 and getting through.